### DISCUSSION TITLE
How to get the transcription grouped by chapters?

### DISCUSSION THREAD

**Discussion by viniciusarruda** [jdepoix/youtube-transcript-api#225] [Q&A]
How to get the transcription grouped by chapters?
Or, an alternative way to get the video chapters range and title.

**Comment by jdepoix** [comment]
Hi @viniciusarruda, this module does not support retrieving chapters. You would have to find some other way to find out the chapters and then group them retrieved transcript by their timestamps.

**Reply by viniciusarruda** [reply]
Do you have any idea on how can I do it?

**Reply by jdepoix** [reply]
I am sorry, I haven't tried to retrieve chapters myself yet, so you'll have to do your own research 😊

**Comment by Gardusio** [comment]
I had the same goal to retrieve chapters titles and timestamps, here's how I've managed to extract chapters :



```python

   #"start" field is in millis you can easily convert
   def extract_chapter_info(obj):
      return {
                "title": obj.get("chapterRenderer").get("title").get("simpleText"),
                "start": obj.get("chapterRenderer").get("timeRangeStartMillis"),
      }
   
   def _extract_chapters_json(self, html, video_id):
        splitted_html = html.split('"chapters":')
        
        # HANDLE NO CHAPTERS SCENARIO AS YOU LIKE
        
        chapters_json = json.loads(
            splitted_html[1].split(',"trackingParams"')[0].replace("\n", "")
        )

        return list(map(extract_chapter_info, chapters_json))
```

This is basically the same of `_extract_captions_json` [here](https://github.com/jdepoix/youtube-transcript-api/blob/99ce79792f943e6dcb85748d40fd73e03e4fe44c/youtube_transcript_api/_transcripts.py#L51)

hope this helps :) don't know if this could become a PR
